google/cloud/bigquery/ipython_magics/line_arg_parser/lexer.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+from collections import OrderedDict
+import itertools
+import re
+
+import enum
+
+
+Token = namedtuple("Token", ("type_", "lexeme", "pos"))
+StateTransition = namedtuple("StateTransition", ("new_state", "total_offset"))
+
+
+# Token definition order is important, thus an OrderedDict is needed with tightly
+# controlled member definitions (i.e. passed as a sequence, and *not* via kwargs).
+token_types = OrderedDict(
+    [
+        (
+            "state_1",
```

Comment:
Can we come up with more descriptive names for these states? For example, is this state finding the argument name?

Reply:
At the time I couldn't think of anything short enough and meaningful, and there are comments in the `LexerState` enum.

Those comments could be easily overlooked when glancing over the code, though, thus I'll try to find more descriptive names.